src/vs/editor/browser/controller/mouseHandler.ts

```diff
@@ -407,12 +415,30 @@ class MouseDownOperation extends Disposable {
 		this._dispatchMouse(position, true);
 	}
 
+	private _onMouseDragThenMove(e: EditorMouseEvent): void {
+		this._lastMouseEvent = e;
+		this._mouseState.setModifiers(e);
+		this._mouseState.setMouseDownEvent('drag');
```

Comment:
`this._mouseState.setMouseDownEvent('drag');` should move to the `start` method when the operation starts. The `lastMouseDownEvent` should be set only once when the operation starts inside `start`.